Document/0x06e-Testing-Cryptography.md

```diff
@@ -2,12 +2,46 @@
 
 In the "Cryptography for Mobile Apps" chapter, we introduced general cryptography best practices and described typical problems that may occur when cryptography is used incorrectly. In this chapter, we'll detail the cryptography APIs available for iOS. We'll show how to identify usage of those APIs in the source code and how to interpret cryptographic configurations. When you're reviewing code, compare the cryptographic parameters with the current best practices linked in this guide.
 
-### iOS Cryptography Libraries
+### Verifying the Configuration of Cryptographic Standard Algorithms
 
+#### Overview
 Apple provides libraries that include implementations of most common cryptographic algorithms. [Apple's Cryptographic Services Guide](https://developer.apple.com/library/content/documentation/Security/Conceptual/cryptoservices/GeneralPurposeCrypto/GeneralPurposeCrypto.html "Apple Cryptographic Services Guide") is a great reference. It contains generalized documentation of how to use standard libraries to initialize and use cryptographic primitives, information that is useful for source code analysis.
 
-iOS code usually refers to constants defined in `CommonCryptor.h` (for example, `kCCAlgorithmDES`). You can search the source code for these constants to detect their use. Because iOS constants are numeric, you should determine whether the constants sent to the `CCCrypt` function represent an insecure or deprecated algorithm.
-
+##### CommonCrypto, SecKeyEncrypt and Wrapper libraries
+The most commonly used Class for cryptographic operations is the CommonCrypto, which is packed with the iOS runtime. The functionality offered by the CommonCrypto object can best be disected by having a look at the [source code of the header file ](https://opensource.apple.com/source/CommonCrypto/CommonCrypto-36064/CommonCrypto/CommonCrypto.h "CommonCrypto.h"):
+- The `Commoncryptor.h` gives the parameters for the symmetric cryptographic operations,
+- The `CommonDigest.h` gives the parameters for the hashing Algorithms
+- The `CommonHMAC.h` gives the parameters for the supported HMAC operations.
+- The `CommonKeyDerivation.h` gives the parameters for supported KDF functions
+- The `CommonSymmetricKeywrap.h` gives the function used for wrappnig a symmetric key with a Key Encryption Key.
+
+CommonCryptor lacks a few type of operations unfortunately in its public APIs, for instance: GCM mode is only available in its private APIs: see [its sourcecode](https://opensource.apple.com/source/CommonCrypto/CommonCrypto-60074/include/CommonCryptorSPI.h "GCM in CC"). For this, an additional binding header is necessary or other wrapper libraries can be used.
```

Comment:
CommonCryptor lacks a few type of operations unfortunately in its public APIs, for instance: ->
Unfortunately, CommonCryptor lacks a few types of operations in its public APIs. For instance:

Reply:
Not sure about this one, perhaps , such as: is better?